frontend/src/components/modalContent/AddIncidentModal.js

```diff
@@ -1,36 +1,284 @@
-import React from "react";
+import React, { Component } from "react";
 import PropTypes from "prop-types";
-import AddIncidentForm from "@/components/Forms/AddIncidentForm";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import { getFormValues, reset } from "redux-form";
+import { Steps, Button, Popconfirm } from "antd";
+import * as FORM from "@/constants/forms";
+import AddIncidentReportingForm from "@/components/Forms/incidents/AddIncidentReportingForm";
+import AddIncidentDetailForm from "@/components/Forms/incidents/AddIncidentDetailForm";
+import AddIncidentFollowUpForm from "@/components/Forms/incidents/AddIncidentFollowUpForm";
+
 import CustomPropTypes from "@/customPropTypes";
 
+const { Step } = Steps;
+
 const propTypes = {
-  closeModal: PropTypes.func.isRequired,
   onSubmit: PropTypes.func.isRequired,
-  title: PropTypes.string,
-  followupActionOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
+  closeModal: PropTypes.func.isRequired,
   incidentDeterminationOptions: CustomPropTypes.options.isRequired,
+  incidentStatusCodeOptions: CustomPropTypes.options.isRequired,
+  followupActionOptions: CustomPropTypes.options.isRequired,
   doSubparagraphOptions: CustomPropTypes.options.isRequired,
   initialValues: PropTypes.objectOf(PropTypes.any).isRequired,
+  inspectors: CustomPropTypes.options.isRequired,
+  addReportingFormValues: PropTypes.objectOf(PropTypes.any),
+  addDetailFormValues: PropTypes.objectOf(PropTypes.any),
+  addFollowUpFormValues: PropTypes.objectOf(PropTypes.any),
+  reset: PropTypes.func.isRequired,
 };
 
 const defaultProps = {
-  title: "",
+  addReportingFormValues: {},
+  addDetailFormValues: {},
+  addFollowUpFormValues: {},
 };
 
-export const AddIncidentModal = (props) => (
-  <div>
-    <AddIncidentForm
-      closeModal={props.closeModal}
-      onSubmit={props.onSubmit}
-      title={props.title}
-      followupActionOptions={props.followupActionOptions}
-      incidentDeterminationOptions={props.incidentDeterminationOptions}
-      doSubparagraphOptions={props.doSubparagraphOptions}
-      initialValues={props.initialValues}
-    />
-  </div>
-);
+// TODO: Should move these into the forms themselves
+
+export class AddIncidentModal extends Component {
```

Comment:
This AddIncidentModal draws heavily on the AddPartyModal patterns. There are some things that might be better abstracted into a resuable component for these 'Wizard' style Forms. Handling conditional rendering for buttons across the seperate forms is being accomplished through nested conditionals on the parent form at the moment - it would be nice to bring it down into the form.